dev-libs/libratbag-0.14: Version Bump (ratbagd + piper)

kwhat opened the pull request:
Bug: https://bugs.gentoo.org/682372
Bug: https://bugs.gentoo.org/629006
Closes: https://bugs.gentoo.org/682372
Closes: https://bugs.gentoo.org/629006
Signed-off-by: Alex Barker alex@1stleg.com

juippis commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,63 @@
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit meson python-r1 udev
```
`python-r1` is most likely wrong, and looks to me like you want to use `python-single-r1` instead.

kwhat replied:
I cannot get this working with python-single-r1.  It looks like dev-python/python-evdev[${PYTHON_USEDEP}] is required. dev-python/python-evdev does not take a singles target, and I cannot use PYTHON_USEDEP with python-single-r1.

kwhat replied:
I think I got it working.

juippis commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,63 @@
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc elogind systemd test"
+REQUIRED_USE="^^ ( systemd elogind ) ${PYTHON_REQUIRED_USE}"
```
Please multiline PYTHON_REQ part, although I doubt it's needed here.

juippis commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,63 @@
+
+DEPEND="
+	${PYTHON_DEPS}
+	virtual/pkgconfig
+	doc? ( app-doc/doxygen )
```
These are both BDEPEND type of dependencies.

kwhat replied:
I also add test to BDEPEND

juippis commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,63 @@
+	doc? ( app-doc/doxygen )
+	elogind? ( sys-auth/elogind )
+	systemd? ( sys-apps/systemd )
+	dev-python/python-evdev[${PYTHON_USEDEP}]
+	dev-lang/swig
```
Please put unconditional deps above conditional ones.

juippis commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,63 @@
+	dev-lang/swig
+	test? ( dev-util/valgrind )
+"
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-libs/libevdev
+	virtual/libudev
```
You want to inherit `${DEPEND}` here also most likely.

juippis commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,63 @@
+
+src_prepare() {
+	default
+
+	sed '/default:/d' -i meson_options.txt || die
```
Add a comment above this sed to explain what it does.

kwhat replied:
If this did something in the past, it doesn't do anything now.  removed.

Arfrever commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,68 @@
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
```
Please also test it with Python 3.8.

kwhat replied:
Unfortunately dev-python/python-evdev[python_targets_python3_8] isn't available.  If there is a way to work around this, let me know.

kwhat replied:
Ok, I added support for python 3.8 and tested.  Everything appears to work fine with all 3 ebuilds.

Arfrever commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,68 @@
+"
+DEPEND="
+	${RDEPEND}
+	dev-lang/swig
```
dev-lang/swig certainly should be in `BDEPEND`.

kwhat replied:
Fixed

Arfrever commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,68 @@
+DEPEND="
+	${RDEPEND}
+	dev-lang/swig
+	$(python_gen_cond_dep 'dev-python/python-evdev[${PYTHON_MULTI_USEDEP}]')
```
At least one installed file `ratbagctl` seems to use it:
https://github.com/libratbag/libratbag/blob/77b67775da6e0a8b2d1f01bee0bcdd94b7d5f68d/tools/ratbagctl.in.in#L24:
```
import evdev
```
So this dependency should be in `RDEPEND`.

kwhat replied:
fixed

Arfrever commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,68 @@
+	dev-lang/swig
+	$(python_gen_cond_dep 'dev-python/python-evdev[${PYTHON_MULTI_USEDEP}]')
+	elogind? ( sys-auth/elogind )
+	systemd? ( sys-apps/systemd )
```
How are elogind and systemd used only at build time and not at run time?

kwhat replied:
fixed

Arfrever commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,68 @@
+	python_setup
+
+	local emesonargs=(
+		-Dsystemd=$(usex systemd true false)
```
For cases with `$(usex ... true false)`, you may consider using shorter variants (`$(meson_use ...)`).
See: https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=b69481722aa911410ab12018f01db859d696f22c

kwhat replied:
Add `meson_use` except for ` -Dlogind-provider=$(usex elogind elogind systemd)`

Arfrever commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,68 @@
+		-Dlogind-provider=$(usex elogind elogind systemd)
+		-Ddocumentation=$(usex doc true false)
+		-Dtests=$(usex test true false)
+		-Dudev-dir=$(get_udevdir)
```
Maybe `${EPREFIX}` is needed: `-Dudev-dir="${EPREFIX}$(get_udevdir)"`

kwhat replied:
Indeed, fixed.

Arfrever commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,68 @@
+IUSE="doc elogind systemd test"
+REQUIRED_USE="
+	${PYTHON_REQUIRED_USE}
+	^^ ( systemd elogind )
```
Maybe alphabetically sorted USE flags (i.e. `^^ ( elogind systemd )`) would look better.

kwhat replied:
Fixed

Arfrever commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,68 @@
+"
+RESTRICT="!test? ( test )"
+
+BDEPEND="
```
Python is used at build time, so `${PYTHON_DEPS}` is needed in `BDEPEND`.
https://github.com/libratbag/libratbag/blob/1097b1ba686e6a0c62868737bc767b34a039b110/meson.build#L541-L547
https://github.com/libratbag/libratbag/blob/1097b1ba686e6a0c62868737bc767b34a039b110/meson.build#L562-L564
https://github.com/libratbag/libratbag/blob/1097b1ba686e6a0c62868737bc767b34a039b110/meson.build#L667-L669

Arfrever replied:
```
tools/ratbagd.py:from gi.repository import Gio, GLib, GObject
tools/toolbox.py:from gi.repository import GLib
tools/toolbox.py:    from gi.repository import Gio
```

`tools/toolbox.py` is used to create `toolbox.py`, which seems to be used only at build time.

`tools/merge_ratbagd.py` seems to create `ratbagctl` (installed) from `'tools/ratbagctl.in.in` and `tools/ratbagd.py`.
So `from gi.repository import Gio, GLib, GObject` is present in installed file.

Assuming that above analysis is correct, both `BDEPEND` and `RDEPEND` need additional dependencies:
- For general `gi` module infrastructure, `dev-python/pygobject:3` is needed (as `$(python_gen_cond_dep 'dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]')`).
- For importing modules specified after `from gi.repository import` (in this case `Gio`, `GLib`, `GObject`), appropriate `/usr/share/gir-1.0/*.gir` files are needed. `dev-libs/gobject-introspection` provides `/usr/share/gir-1.0/Gio-2.0.gir`, `/usr/share/gir-1.0/GObject-2.0.gir`, `/usr/share/gir-1.0/GLib-2.0.gir`.

kwhat replied:
Fixed for both ebuilds. 

- Python is also required by piper at build time.

Arfrever commented on dev-libs/libratbag/metadata.xml:
```diff
@@ -1,8 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+
```
This empty line seems non-standard...
(The same in ` app-misc/piper/metadata.xml`.)

kwhat replied:
Fixed both ebuilds.

Arfrever commented on app-misc/piper/piper-0.4.ebuild:
```diff
@@ -0,0 +1,44 @@
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
```
Please also test it with Python 3.8.

kwhat replied:
Same issue with python 3.8 as libratbag.

kwhat replied:
Resolved.

Arfrever commented on app-misc/piper/piper-0.4.ebuild:
```diff
@@ -0,0 +1,44 @@
+	${PYTHON_DEPS}
+	dev-libs/libratbag
+	$(python_gen_cond_dep 'dev-python/python-evdev[${PYTHON_MULTI_USEDEP}]')
+	$(python_gen_cond_dep 'dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]')
```
Additional dependencies are needed:
```
$ grep -Er '(gi\.require_version|from gi\.repository import)' *
piper/application.py:gi.require_version("Gio", "2.0")
piper/application.py:gi.require_version("Gtk", "3.0")
piper/application.py:from gi.repository import Gio, GLib, Gtk  # noqa
piper/buttondialog.py:gi.require_version("Gtk", "3.0")
piper/buttondialog.py:from gi.repository import Gdk, GObject, Gtk  # noqa
piper/buttonspage.py:gi.require_version("Gtk", "3.0")
piper/buttonspage.py:from gi.repository import Gtk  # noqa
piper/devicerow.py:gi.require_version("Gtk", "3.0")
piper/devicerow.py:from gi.repository import GdkPixbuf, GObject, Gtk, Rsvg  # noqa
piper/errorperspective.py:gi.require_version("Gtk", "3.0")
piper/errorperspective.py:from gi.repository import GObject, Gtk  # noqa
piper/gi_composites.py:from gi.repository import Gio, GLib, GObject, Gtk  # noqa
piper/leddialog.py:gi.require_version("Gtk", "3.0")
piper/leddialog.py:from gi.repository import Gdk, GObject, Gtk  # noqa
piper/ledspage.py:gi.require_version("Gtk", "3.0")
piper/ledspage.py:from gi.repository import Gtk  # noqa
piper/mousemap.py:gi.require_version("Gdk", "3.0")
piper/mousemap.py:gi.require_version("Gtk", "3.0")
piper/mousemap.py:gi.require_version("Rsvg", "2.0")
piper/mousemap.py:from gi.repository import Gdk, GLib, Gtk, GObject, Rsvg  # noqa
piper/mouseperspective.py:gi.require_version("Gtk", "3.0")
piper/mouseperspective.py:from gi.repository import GLib, GObject, Gtk  # noqa
piper/optionbutton.py:gi.require_version("Gtk", "3.0")
piper/optionbutton.py:from gi.repository import Gtk  # noqa
piper/profilerow.py:gi.require_version("Gtk", "3.0")
piper/profilerow.py:from gi.repository import GObject, Gtk  # noqa
piper/ratbagd.py:from gi.repository import Gio, GLib, GObject  # noqa
piper/resolutionrow.py:gi.require_version("Gtk", "3.0")
piper/resolutionrow.py:from gi.repository import GObject, Gtk  # noqa
piper/resolutionspage.py:gi.require_version("Gtk", "3.0")
piper/resolutionspage.py:from gi.repository import Gtk  # noqa
piper/svg.py:from gi.repository import Gio  # noqa
piper/welcomeperspective.py:gi.require_version("Gtk", "3.0")
piper/welcomeperspective.py:from gi.repository import GObject, Gtk, Gdk  # noqa
piper/window.py:gi.require_version("Gtk", "3.0")
piper/window.py:from gi.repository import Gdk, GLib, Gtk, Gio  # noqa
piper.in:gi.require_version('Gio', '2.0')
piper.in:gi.require_version('Gtk', '3.0')
piper.in:from gi.repository import Gio, Gtk
```

Many packages install GObject-Introspection files only conditionally with USE="introspection", so `[introspection]` will be needed for them.
In this case:
- `gnome-base/librsvg[introspection]` is needed for `/usr/share/gir-1.0/Rsvg-2.0.gir`.
- `x11-libs/gdk-pixbuf[introspection]` is needed for `/usr/share/gir-1.0/GdkPixbuf-2.0.gir`.
- `x11-libs/gtk+:3[introspection]` is needed for `/usr/share/gir-1.0/Gdk-3.0.gir` and `/usr/share/gir-1.0/Gtk-3.0.gir`.
- You may add `dev-libs/gobject-introspection` for `Gio`, `GLib` and `GObject`, although it is guaranteed to be installed by other dependencies...

Above dependencies are needed at least in `RDEPEND`.

Arfrever replied:
https://github.com/libratbag/piper/blob/0ab5c63ed404f619dcd6c380d8881d3f9eb3d406/meson.build#L73 and https://github.com/libratbag/piper/blob/65c740ced2960548b286b7beaf49a2a798f36be3/piper/mousemap.py#L20 say that `dev-python/lxml` is needed.

https://github.com/libratbag/piper/blob/0ab5c63ed404f619dcd6c380d8881d3f9eb3d406/meson.build#L75 and https://github.com/libratbag/piper/blob/65c740ced2960548b286b7beaf49a2a798f36be3/piper/mousemap.py#L17 say that `dev-python/pycairo` is needed.

kwhat replied:
Glad you pointed out [introspection], I missed then when trying to find these missing depends on my own.  Fixed.

Arfrever commented on app-misc/piper/piper-0.4.ebuild:
```diff
@@ -0,0 +1,44 @@
+"
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-libs/libratbag
```
https://github.com/libratbag/piper/blob/0ab5c63ed404f619dcd6c380d8881d3f9eb3d406/meson.build#L16-L17 says that `>=dev-libs/libratbag-0.13` is needed.

kwhat replied:
Correct, fixed.

kwhat commented:
@Arfrever Thanks for the code review. I have addressed all review items and marked the fully resolved items as such.  The two remaining items are `meson_use`  for -Dlogind-provider=$(usex elogind elogind systemd) which doesn't look possible and the python3.8 testing is also blocked by `dev-python/python-evdev[python_targets_python3_8]`.

Arfrever commented:
About `dev-python/python-evdev`, you can locally change it to add support for Python 3.8, and test all 3 packages (`dev-python/python-evdev`, `dev-libs/libratbag`, `app-misc/piper`) with Python 3.8. If they work, you can add additional commit for `dev-python/python-evdev`, which would only add support for Python 3.8.

Arfrever commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -30,14 +32,13 @@ RDEPEND="
-DEPEND="
-	${RDEPEND}
-	dev-lang/swig
+	$(python_gen_cond_dep 'dev-libs/gobject-introspection[${PYTHON_SINGLE_USEDEP}]')
```
If depending on `dev-libs/gobject-introspection`, in this case, Python modules provided by `dev-libs/gobject-introspection` are not used, so drop `$(python_gen_cond_dep '` and `[${PYTHON_SINGLE_USEDEP}]')`.

Arfrever commented on app-misc/piper/piper-0.4.ebuild:
```diff
@@ -18,13 +18,20 @@ IUSE=""
-	$(python_gen_cond_dep 'dev-python/python-evdev[${PYTHON_MULTI_USEDEP}]')
+	>=dev-libs/libratbag-0.13
+	dev-python/lxml
+	dev-python/pycairo
```
Modules from `dev-python/lxml` and `dev-python/pycairo` are imported, so for both of them, `$(python_gen_cond_dep ...` is needed.

Arfrever commented on app-misc/piper/piper-0.4.ebuild:
```diff
@@ -18,13 +18,20 @@ IUSE=""
+	gnome-base/librsvg[introspection]
+	x11-libs/gdk-pixbuf[introspection]
+	x11-libs/gtk+:3[introspection]
+	$(python_gen_cond_dep 'dev-libs/gobject-introspection[${PYTHON_SINGLE_USEDEP}]')
```
`$(python_gen_cond_dep '` and `[${PYTHON_SINGLE_USEDEP}]')` should be dropped.
`giscanner` module is not used by either `dev-libs/libratbag` or `app-misc/piper`.
`giscanner` module is installed in custom directory (`/usr/lib64/gobject-introspection/giscanner`) instead of in `site-packages`, and is rather only for private usage by scripts from `dev-libs/gobject-introspection` (`/usr/bin/g-ir-annotation-tool`, `/usr/bin/g-ir-scanner`).

Arfrever commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -17,11 +17,13 @@ KEYWORDS="~amd64 ~x86"
+	dev-lang/swig
 	virtual/pkgconfig
 	doc? ( app-doc/doxygen )
 	test? ( dev-util/valgrind )
```
For better accuracy in case of crosscompilation, this should be:
```
	test? (
		dev-libs/gobject-introspection
		dev-util/valgrind
		$(python_gen_cond_dep 'dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]')
		$(python_gen_cond_dep 'dev-python/python-evdev[${PYTHON_MULTI_USEDEP}]')
	)
```
https://github.com/libratbag/libratbag/blob/1097b1ba686e6a0c62868737bc767b34a039b110/meson.build#L588-L596 creates `ratbagctl.test` from `tools/ratbagctl.test.in`.
https://github.com/libratbag/libratbag/blob/f82c6f72b692af29b1f220b2e04061eec9202811/tools/ratbagctl.test.in#L32 imports `toolbox.py` created in https://github.com/libratbag/libratbag/blob/1097b1ba686e6a0c62868737bc767b34a039b110/meson.build#L581-L583 from `tools/toolbox.py`.
And https://github.com/libratbag/libratbag/blob/f82c6f72b692af29b1f220b2e04061eec9202811/tools/toolbox.py#L29 and https://github.com/libratbag/libratbag/blob/f82c6f72b692af29b1f220b2e04061eec9202811/tools/toolbox.py#L110-L112 import other modules...

kwhat commented:
I hope I didn't miss anything.  All changes were squashed up and I bumped evdev-python to 1.3.0.

juippis commented on app-misc/piper/piper-0.4.ebuild:
```diff
@@ -0,0 +1,51 @@
+	gnome-base/librsvg[introspection]
+	x11-libs/gdk-pixbuf[introspection]
+	x11-libs/gtk+:3[introspection]
+	$(python_gen_cond_dep 'dev-python/lxml[${PYTHON_MULTI_USEDEP}]')
+	$(python_gen_cond_dep 'dev-python/pycairo[${PYTHON_MULTI_USEDEP}]')
+	$(python_gen_cond_dep 'dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]')
+	$(python_gen_cond_dep 'dev-python/python-evdev[${PYTHON_MULTI_USEDEP}]')
```
```
$(python_gen_cond_dep '
	dev-python/lxml[${PYTHON_MULTI_USEDEP}]
	dev-python/pycairo[${PYTHON_MULTI_USEDEP}]
	dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]
	dev-python/python-evdev[${PYTHON_MULTI_USEDEP}]
')
```

juippis commented on dev-python/python-evdev/Manifest:
```diff
@@ -1 +1 @@
-DIST python-evdev-1.1.2.tar.gz 83067 BLAKE2B 6434d5c62e4a072928d555bfc92e94e7f63ef98ebbf9cd95795a92d6ccdf7b5d4b0f3133a4b3f2524083b842f78ccc305a5da13af59052be3724d83ea4039328 SHA512 054b2e5af532079dc52919d90791d12d7caf75cade07d18b94ea519869659413ea059570b16ebb236a715c541b17016db380bea4e3c1cd21f96824c6303b45d6
+DIST python-evdev-1.3.0.tar.gz 85018 BLAKE2B ba5c391631ecf8cc0a46f21830209c59b1217940ff25e051e027bc8896d71965a394f222f4e3d629ba2a2f8056ccbdc0dfaf43a37684d7157c7456d61f2d8314 SHA512 392fb4fabed344e6db4e257d4cfc7043393efda15d940be80b013e4c4e350334148a7b04aa0a7239f824ad2c7e0bb057c5cff1c0f320d30b25b0c9aab65a6891
```
You can't do a version bump like this. You need to make one commit for adding a new file, and 2nd commit for removing one.

juippis commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,74 @@
+	dev-libs/gobject-introspection
+	dev-libs/libevdev
+	virtual/libudev
+	$(python_gen_cond_dep 'dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]')
+	$(python_gen_cond_dep 'dev-python/python-evdev[${PYTHON_MULTI_USEDEP}]')
```
merge into one `python_gen_cond_dep` as shown earlier.

juippis commented on dev-libs/libratbag/libratbag-0.13.ebuild:
```diff
@@ -0,0 +1,74 @@
+	test? (
+		dev-libs/gobject-introspection
+		dev-util/valgrind
+		$(python_gen_cond_dep 'dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]')
+		$(python_gen_cond_dep 'dev-python/python-evdev[${PYTHON_MULTI_USEDEP}]')
```
Same here.

thesamesam commented on app-misc/piper/piper-0.4.ebuild:
```diff
@@ -0,0 +1,51 @@
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
Drop

olifre commented:
@kwhat Since `dev-python/python-evdev` with 3.8 support is now in tree, I think the remaining CI error(s) should go away once you rebase this PR onto current master.

juippis reviewed (commented):
Sorry looks like I've lost this PR. On it now.

```
meson.build:366:1: ERROR: Dependency "check" not found, tried pkgconfig and cmake

A full log can be found at /var/tmp/portage/gpm/portage/dev-libs/libratbag-0.14/work/libratbag-0.14-build/meson-logs/meson-log.txt
 * ERROR: dev-libs/libratbag-0.14::testworld failed (configure phase):
 *   (no error message)
```
I also see upstream has made a new release, could you update to latest release and check the updated deps?
https://github.com/libratbag/libratbag/blob/f7c22ed88c31f03e6beddd6f04d3f7826ec3b7e2/meson.build ie looks like check is needed with `test` USE flag.

juippis reviewed (commented):
```
Run-time dependency json-glib-1.0 found: NO (tried pkgconfig and cmake)

meson.build:75:0: ERROR: Dependency "json-glib-1.0" not found, tried pkgconfig and cmake

A full log can be found at /var/tmp/portage/gpm/portage/dev-libs/libratbag-0.15/work/libratbag-0.15-build/meson-logs/meson-log.txt
 * ERROR: dev-libs/libratbag-0.15::testworld failed (configure phase):
 *   (no error message)
```

I asked you to go through their listed dependencies...

https://github.com/libratbag/libratbag/blob/f7c22ed88c31f03e6beddd6f04d3f7826ec3b7e2/meson.build ctrl+f `dependency(` shows at least 2 deps not listed in here, and I didn't go through them all.

jsmolic commented:
This should also solve gcc-10 build issues. Please tag https://bugs.gentoo.org/707440 as well.\
And https://bugs.gentoo.org/723030, it should depend on dev-libs/check for tests
